flocker/node/functional/test_deploy.py

```diff
@@ -280,13 +277,44 @@ def test_links_lowercase(self):
             NodeState(hostname=deployer.hostname, uuid=deployer.node_uuid,
                       applications=[], used_ports=[],
                       manifestations={}, paths={}, devices={})))
+        return d
+
+    @if_docker_configured
+    def test_links_lowercase(self):
+        """
+        Lower-cased link aliases do not result in lack of covergence.
+
+        Environment variables introspected by the Docker client for links
+        are all upper-case, a source of potential problems in detecting
+        the state.
+        """
+        link = Link(alias=u"alias",
+                    local_port=80,
+                    remote_port=8080)
+        d = self._start_container_for_introspection(
+            links=[link],
+            command_line=[u"nc", u"-l", u"-p", u"8080"])
         d.addCallback(
             lambda results: self.assertIn(
-                application.links,
+                pset([link]),
                 [app.links for app in results[0].applications]))
         return d
 
     @if_docker_configured
+    def test_command_line_introspection(self):
+        """
+        Checking the command-line status results in same command-line we
+        passed in.
+        """
+        command_line = pvector([u"nc", u"-l", u"-p", u"8080"])
+        d = self._start_container_for_introspection(command_line=command_line)
+        d.addCallback(
+            lambda results: self.assertIn(
+                command_line,
+                [app.command_line for app in results[0].applications]))
```

Comment:
:question: There should only be *one* application, right? So maybe I'd unpack the applications into `[application]` and use `assertEqual`.